tracer.go

```diff
@@ -121,6 +126,110 @@ type StartSpanOptions struct {
 	// identical to those for Span.SetTag(). May be nil.
 	//
 	// If specified, the caller hands off ownership of Tags at
-	// StartSpanWithOptions() invocation time.
+	// StartSpan() invocation time.
 	Tags map[string]interface{}
 }
+
+// StartSpanOption instances (zero or more) may be passed to Tracer.StartSpan.
+type StartSpanOption interface {
+	Apply(*StartSpanOptions)
+}
+
+// ReferenceType is an enum type describing different categories of
+// relationships between two Spans. If Span-2 refers to Span-1, the
+// ReferenceType describes Span-1 from Span-2's perspective. For example,
+// RefBlockedParent means that Span-1 caused Span-2, and that it's blocked on
+// Span-2's finish.
+type ReferenceType int
```

Comment:
The references can also be used for _meta-trace_ linkage, e.g. an http request (trace-1) enqueuing a background job (trace-2). Can we clarify in the comments that span-1 and span-2 do not have to belong to the same trace?